cicecore/cicedynB/dynamics/ice_transport_driver.F90

```diff
@@ -1602,8 +1612,13 @@ subroutine state_to_work (nx_block, ny_block,        &
                                         * trcrn(i,j,it     ,n)
                enddo
                enddo
+#ifdef UNDEPRECATE_CESMPONDS
             elseif (trcr_depend(it) == 2+nt_apnd .and. &
                     tr_pond_cesm .or. tr_pond_topo) then
+#else
```

Comment:
This is interesting.  I wonder whether the .and. or the .or. took precedence in the logic for this elseif statement.
i.e.
`(trcr_depend(it)==2+nt_apnd .and. tr_pond_cesm) .or. tr_pond_topo`
or
`trcr_depend(it)==2+nt_apnd .and. (tr_pond_cesm .or. tr_pond_topo)`
?
The latter is correct, but what was the code actually doing?

Reply:
According to https://www.intel.com/content/www/us/en/develop/documentation/fortran-compiler-oneapi-dev-guide-and-reference/top/language-reference/expressions-and-assignment-statements/expressions/summary-of-operator-precedence.html the `.and.` would have precedence on the `.or.` so the code would do the first thing you listed:
```fortran
(trcr_depend(it)==2+nt_apnd .and. tr_pond_cesm) .or. tr_pond_topo
```
i.e. the wrong thing if I understand correctly.

Reply:
Thanks.  This line is in the state_to_work routine, which is only used for upwind advection.  Tracers may not work properly with upwind advection -- see https://github.com/CICE-Consortium/CICE/issues/267 and https://github.com/CICE-Consortium/CICE/issues/542. This faulty logic could be part of the reason why, but it probably doesn't fix the entire issue, which is that many tracers are not implemented at all for upwind.